src/test/java/com/amplitude/AmplitudeTest.java

```diff
@@ -17,4 +40,99 @@ public void testGetInstance() {
     assertNotSame(d, e);
     assertNotSame(a, d);
   }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testLogEventSuccess(boolean useBatch)
+      throws InterruptedException, NoSuchFieldException, IllegalAccessException,
+          AmplitudeInvalidAPIKeyException {
+    Amplitude amplitude = Amplitude.getInstance("test");
+    amplitude.init(apiKey);
+    amplitude.useBatchMode(useBatch);
+    amplitude.setLogMode(AmplitudeLog.LogMode.OFF);
+    List<Event> events = EventsGenerator.generateEvents(10, 5, 6);
+    HttpCall httpCall = getMockHttpCall(amplitude, useBatch);
+    Response response = new Response();
+    response.code = 200;
+    response.status = Status.SUCCESS;
+    CountDownLatch latch = new CountDownLatch(1);
```

Comment:
I'm sure this is right, but please explain to me. Why are we waiting when the response is mocked?

Reply:
yeah, the issue here is for our call is inside the async or another thread, so even if response is called, there is no way to guarantee our assertion called after the response is returned and all the async part finished. So in this case we can also do `Thread.sleep` in tests, but the waiting time getting tricky, a long time is fine but getting the tests to always taking a longer time, a short time could make the tests flakey, which is not ideal. So just use the CountDownLatch here to await and make sure our assertion happens after the call to prevent the sleep part.

Reply:
Ohhh, that makes a lot of sense. That's really deep async logic. Didn't actually remember this despite the fact I did work on the async calls of Java SDK. This `CountDownLatch` is probably the cleanest way to handle this.